app/controllers/v0/sessions_controller.rb

```diff
@@ -19,37 +15,19 @@ class SessionsController < ApplicationController
     # Collection Action: auth is required for certain types of requests
     # @type is set automatically by the routes in config/routes.rb
     # For more details see SAML::SettingsService and SAML::URLService
-    # rubocop:disable Metrics/CyclomaticComplexity, Metrics/MethodLength
     def new
-      url = case params[:type]
-            when 'mhv'
-              reset_session
-              url_service.mhv_url
-            when 'dslogon'
-              reset_session
-              url_service.dslogon_url
-            when 'idme'
-              reset_session
-              url_service.idme_loa1_url + (params[:signup] ? '&op=signup' : '')
-            when 'mfa'
-              authenticate
-              url_service.mfa_url
-            when 'verify'
-              authenticate
-              url_service.idme_loa3_url
-            when 'slo'
-              authenticate
-              logout_url = url_service.slo_url
-              Rails.logger.info('SSO: LOGOUT', sso_logging_info)
-              reset_session
-              logout_url
-            end
-      respond_to do |format|
-        format.html { redirect_to url }
-        format.json { render json: { url: url } }
+      type = params[:signup] ? 'signup' : params[:type]
+      if REDIRECT_URLS.include?(type)
+        url = url_service.send("#{type}_url")
+        if type == 'slo'
+          Rails.logger.info('SSO: LOGOUT', sso_logging_info)
+          reset_session
+        end
+        redirect_to url
```

Comment:
@narin i would just tack the clientID on the end of URL here, no point in going through URLService.

Reply:
👍🏼 I'll wait for this PR to get merged and then tack on the clientID.